### ISSUE TITLE
d_store

### ISSUE THREAD

**Issue by janbiedermann** [isomorfeus/isomorfeus-framework#30]
Back to to tech, we have client local state, app_storeetc, and shared 'state', with i-data, graph/models etc.
What if we could have a state, that is shared across clients? Not shure how to call it, maybe d_store for d_istributed store. Sure, it needs some access restrictions, maybe share the state across a limited set of clients, based on user role or so. Or based on some other object thats shared.
For example, lets say Joe (from :flagge-us:) and Li (from :flagge-cn:) and Igor (from :flagge-ru:) work a graph they share, together, making changes. Maybe they want to chat, ok, we have channels for that, but maybe they also want to share some data concerning the current context, the graph in our case, which is maybe not meant to be in the graph (yet). For example working on some larger text, before it becomes part of the graph or so.
Now we could attach with some dsl the d_store:
class MyGraph
   use_d_store { |graphinstance| graph_instance.key } # using a identifier for restricting the store to instances
          # based on graph.key (which in rails terms would be something.id)
Then in the graph instance that shared by Joe, Li and Igor, identified by graph.key, when Joe does:
my_graph.d_store.shared_text = "Super important"
this will then be accessible by Li and Igor:
my_graph.d_store.shared_text
Basically everybody using the instance could access that now.
But, we can use policy on the client, so we can restrict that. Not sure yet how exactly that could look like in code. Also the server would have that information. (bearbeitet) 

The idea is, to make d_store available for any class/instance, whatever it is.
